prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -536,12 +552,40 @@ void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
 				size_t data_size)
 {
 	struct rxr_rx_entry *rx_entry;
+	ssize_t ret;
 
 	rx_entry = pkt_entry->x_entry;
 	assert(rx_entry);
 	rx_entry->bytes_copied += data_size;
 
 	if (rx_entry->total_len == rx_entry->bytes_copied) {
+		/*
+		 * rxr_cq_handle_rx_completion() releases pkt_entry, thus
+		 * we do not release it here.
+		 */
+		if (rxr_get_base_hdr(pkt_entry->pkt)->flags & RXR_REQ_WAIT_RECEIPT) {
+		    rx_entry->tx_id = rxr_get_dc_rtm_base_hdr(pkt_entry->pkt)->tx_id;
+			rx_entry->msg_id = rxr_get_rtm_base_hdr(pkt_entry->pkt)->msg_id;
+			rx_entry->addr = pkt_entry->addr;
```

Comment:
Not sure the tx_id, msg_id and addr need to be set here. If anything, they need to be set during initialization of rx_entry.

Reply:
They are not needed, I have removed these assignments. Mark as resolved.